engine/apps/webhooks/tests/test_trigger_webhook.py

```diff
@@ -130,51 +131,88 @@ def test_execute_webhook_integration_filter_matching(
     )
 
 
+ALERT_GROUP_PUBLIC_PRIMARY_KEY = "IXJ47FKMYYJ5U"
+
+
+@httpretty.activate(verbose=True, allow_net_connect=False)
```

Comment:
the exception I was trying to fix, and fully test, here happens rather deep inside the `requests` package ([logs](https://ops.grafana-ops.net/goto/SrFNV8tIR?orgId=1)). Rather than trying to figure out what to properly mock there I decided to try out `httpretty` which according to their [docs](https://httpretty.readthedocs.io/en/latest/introduction.html#motivation):

> It can be a bit of a hassle to use something like mock.Mock to stub the requests, this can work well for low-level unit tests but when writing functional or integration tests we should be able to allow the http calls to go through the TCP socket module.
> 
> HTTPretty [monkey patches](http://en.wikipedia.org/wiki/Monkey_patch) Python’s [socket](https://docs.python.org/3/library/socket.html#module-socket) core module with a fake version of the module.
> 
> Because HTTPretty implements a fake the modules [socket](https://docs.python.org/3/library/socket.html#module-socket) and [ssl](https://docs.python.org/3/library/ssl.html#module-ssl) you can use write tests to code against any HTTP library that use those modules.

in other words, we don't have to patch `requests` at all (**note**: I'm `patch`ing `requests` but w/ the `wraps` kwarg which basically sets a "spy" on the package and doesn't mock its actual implementation ([docs](https://docs.python.org/3/library/unittest.mock.html)))